src/main/java/doggytalents/common/talent/SwimmerDogTalent.java

```diff
@@ -56,4 +97,96 @@ public InteractionResultHolder<Integer> determineNextAir(AbstractDog dogIn, int
 
         return InteractionResultHolder.pass(currentAir);
     }
+
+    @Override
+    public InteractionResult canSwimUnderwater(AbstractDog dogIn) {
+        return InteractionResult.SUCCESS;
+    }
+
+    
+    public static class SwimmerDogGoal extends Goal {
+
+        private Dog dog;
+        private boolean isGettingAir = false;
+        private SmoothSwimmingMoveControl moveControl;
+        private WaterBoundPathNavigation navigator;
+
+        public SwimmerDogGoal(Dog d) {
+            this.dog = d;
+            this.moveControl = new SmoothSwimmingMoveControl(d, d.getMaxHeadXRot(), d.getMaxHeadYRot(), 1, 1, false);
+            this.navigator = new WaterBoundPathNavigation(d, d.level);
+            this.setFlags(EnumSet.of(Flag.JUMP));
+        }
+        
+        @Override
+        public boolean canUse() {
+            return (this.dog.isInWater() || this.isGettingAir) && this.dog.canSwimUnderwater() && !this.checkSurroundingForLand(this.dog, this.dog.blockPosition());
+        }
+
+        @Override
+        public boolean canContinueToUse() {
+            return this.canUse();
+        }
+
+        @Override
+        public void start() {
+            this.dog.setNavigation(this.navigator);
+            this.dog.setMoveControl(this.moveControl);
+            if (this.dog.isInSittingPose()) {
+                this.dog.setInSittingPose(false);
+            }
+            this.applySwimAttributes();
+            this.dog.setPathfindingMalus(BlockPathTypes.WATER_BORDER, 0);
+            this.dog.setPathfindingMalus(BlockPathTypes.WATER, 0);
+        }
+
+        @Override 
+        public void tick() {
```

Comment:
The tick object can be divided into three part the first part would check if the dog enters lowAir mode and trigger a boolean value and perform setup for the float, the second part carries out the float and the final part undo the first part if the dog reach high air supply.